packages/daimo-api/src/db/notifications.ts

```diff
@@ -0,0 +1,86 @@
+import EventEmitter from "node:events";
+import { Client, ClientConfig } from "pg";
+
+const IGNORED_EVENTS = new Set(["newListener", "removeListener"]);
+
+/**
+ * Connects to Postgres and emits LISTEN/NOTIFY events.
+ */
+export class DBNotifications extends EventEmitter {
+  client: Client;
+  channels: Set<string>;
+  isConnected = false;
+
+  constructor(config: ClientConfig) {
+    super();
+    this.client = new Client(config);
+    this.channels = new Set();
+
+    this.on("newListener", (event, listener) => {
+      if (IGNORED_EVENTS.has(event)) {
+        return;
+      }
+
+      if (!this.channels.has(event)) {
+        this.#addChannel(event);
+      }
+    });
+
+    this.on("removeListener", (event, listener) => {
+      if (IGNORED_EVENTS.has(event)) {
+        return;
+      }
+
+      if (this.listenerCount(event) === 0) {
+        this.#removeChannel(event);
+      }
+    });
+  }
+
+  async close() {
+    if (this.client) {
+      await this.client.end();
+
+      this.channels.clear();
+    }
+  }
+
+  async #setupListener() {
```

Comment:
how does `#method` work vs `private method` ?

Reply:
private method is typescript the compiler thing, #method is javascript the language thing. the latter came later that's why we have both of them in ts now. also, #method make sure method is private in the runtime. 

can move to `private method` for consistency across code base.

Reply:
Done in a1a561f970c8c5d7ac0f0cad889d8de815785ce7